src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -99,7 +99,6 @@ class OrderedRVD(
 
     require(ordType.rowType == typ.rowType)
     require(ordType.kType isPrefixOf typ.kType)
-    require(newPartitioner.pkType isIsomorphicTo ordType.pkType)
```

Comment:
Why is this no longer required?

Reply:
Patrick can explain it better, but the idea is that a something partitioned by {locus, alleles} can be joined with something partitioned by {locus} (as {locus, alleles} is a subtype of {locus}). My table key changes in Python caused problems to pop up that were previously being masked by the horribly inefficient join strategy (re-key every time). This is the solution to that.

Reply:
@patrick-schultz is there a different requirement we should check here?

Reply:
I've been trying to build the join infrastructure so that there is no requirement on the partition keys whatsoever (and I think eventually they will go away). I think https://github.com/hail-is/hail/pull/3442/files#diff-cc90b5eede7c86a868dd954853660e37L40 enabled this to be removed, and I just missed it.